lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4373,6 +4443,8 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
           assert(adjointSuccBB && adjointSuccBB->getNumArguments() == 1);
           SILBuilder adjointTrampolineBBBuilder(adjointSuccBB);
           SmallVector<SILValue, 8> trampolineArguments;
+          // Propagate pullback struct argument.
+          trampolineArguments.push_back(adjointSuccBB->getArguments().front());
```

Comment:
Why make the pullback struct argument be the last? This would revert a change I just made in #25255 but it is a bit unclear why it's necessary.

Reply:
Reverted the pullback struct argument reordering in https://github.com/apple/swift/pull/25509/commits/7459d4ec1c0a2af3c03ca7cfac78c666e14adce9. Originally I ran into some crash without the reordering, but I can't reproduce it anymore. Reverting is more efficient (not creating/propagating pullback struct argument if there are no dominated BB active values).

[Related comment here](https://github.com/apple/swift/pull/25509#discussion_r294102117).